lib/form_data.js

```diff
@@ -87,31 +89,40 @@ FormData.prototype._trackLength = function(header, value) {
   });
 };
 
-FormData.prototype._multiPartHeader = function(field, value) {
+FormData.prototype._multiPartHeader = function(field, value, options) {
   var boundary = this.getBoundary();
-  var header =
-    '--' + boundary + FormData.LINE_BREAK +
-    'Content-Disposition: form-data; name="' + field + '"';
-
-  // fs- and request- streams have path property
-  // TODO: Use request's response mime-type
-  if (value.path) {
-    header +=
-      '; filename="' + path.basename(value.path) + '"' + FormData.LINE_BREAK +
-      'Content-Type: ' + mime.lookup(value.path);
-
-  // http response has not
-  } else if (value.readable && value.hasOwnProperty('httpVersion')) {
-    header +=
-      '; filename="' + path.basename(value.client._httpMessage.path) + '"' + FormData.LINE_BREAK +
-      'Content-Type: ' + value.headers['content-type'];
+  var header = '';
+
+  // custom header specified (as string)?
+  // it becomes responsible for boundary
+  // (e.g. to handle extra CRLFs on .NET servers)
+  if (options.header != null) {
```

Comment:
I'd have to recreate the whole `Content-Disposition`, `Content-Type`, `name`, `filename`, for that I'd have to add mime as direct dependency and basically copy most of `_multiPartHeader `, this doesn't make sense.

Isn't this much clearer this way ?
```js
var FormData = require('form-data');
var form = new FormData();

var options = {
  header: {
    'X-Custom-Header': 123
  }
};

form.append('my_thing', theThing, options);
```